packages/app/src/cli/services/environment.ts

```diff
@@ -69,6 +59,58 @@ interface DevEnvironmentOutput {
   tunnelPlugin: string | undefined
 }
 
+/**
+ * Make sure there is a valid environment to execute `generate extension`
+ *
+ * We just need a valid app API key to access the Specifications API.
+ * - If the API key is provided via flag, we use it.
+ * - Else, if there is cached API key for the current directory, we use it.
+ * - Else, we prompt the user to select/create an app.
+ *
+ * The selection is then cached as the "dev" app for the current directory.
+ */
+export async function ensureGenerateEnvironment(options: {
+  apiKey?: string
+  directory: string
+  reset: boolean
+  token: string
+}): Promise<string> {
+  if (options.apiKey) {
+    const app = await fetchAppFromApiKey(options.apiKey, options.token)
+    if (!app) throw InvalidApiKeyError(options.apiKey)
```

Comment:
I'd have expected ESLint to fail this one. Let's abort from here and move away from the factory function pattern for creating errors.

Reply:
Yep, will change it

Reply:
I think the rule for non factory errors doesn't work in all cases 🤔